test_vm/src/lib.rs

```diff
@@ -51,7 +51,7 @@ pub use messaging::*;
 
 /// An in-memory rust-execution VM for testing builtin-actors that yields sensible stack traces and debug info
 pub struct TestVM {
-    pub primitives: FakePrimitives,
+    pub primitives: RefCell<FakePrimitives>,
```

Comment:
If this is going to be boxed, can we use `RefCell<dyn Primitives`> instead? It would be much nicer to be able to `set_primitives(...)` rather than expose a method for each one individually.

Reply:
Currently relying on FakePrimitives being Cloneable to pass out a Primitives instance. 

dyn Primitives wouldn't be cloneable. Though in theory all Primitive implementations should be cloneable (real FVM implementation is just a shim to SDK functionality and fakes have cloneable function pointers) it makes it not object safe. 

A workaround is to add an explicit "clone" method on the primitives trait that returns a Box<Primitive>

Reply:
set_primitives should be able to just take a FakePrimitives though

Reply:
Another option is for Primitives to break the interior-mutability pattern and require a mutable instance to set_primitives. This avoids having the RefCell allowing us to revert to returning a &Primitives (which means we don't have to clone)